### PR TITLE
Make updateSelectizeInput() work with labels again

### LINKED ISSUE #2245: updateSelectizeInput label is broken in shiny 1.2
updateSelectizeInput changing the label does not work in shiny 1.2.  Please see example:

```
#
# This is a Shiny web application. You can run the application by clicking
# the 'Run App' button above.
#
# Find out more about building applications with Shiny here:
#
#    http://shiny.rstudio.com/
#

library(shiny)

# Define UI for application that draws a histogram
ui <- fluidPage(
   useShinyjs(),
   # Application title
   titlePanel("Old Faithful Geyser Data"),
   # Sidebar with a slider input for number of bins 
   sidebarLayout(
      sidebarPanel(
         sliderInput("bins",
                     "Number of bins:",
                     min = 1,
                     max = 50,
                     value = 30),
         actionButton('testb','Button'),
         selectizeInput("test","Select",choices=c("Apple","Banana","Pineapple"),options=list(create=T))
      ),
      
      # Show a plot of the generated distribution
      mainPanel(
         plotOutput("distPlot")
      )
   )
)

# Define server logic required to draw a histogram
server <- function(input, output,session) {
  
  observeEvent(input$testb,{
    updateSelectizeInput(session,"test",label="SelectMore",choices=c("Pinecone","Mashmallow","Candy"))
  })
   
   output$distPlot <- renderPlot({
      # generate bins based on input$bins from ui.R
      x    <- faithful[, 2] 
      bins <- seq(min(x), max(x), length.out = input$bins + 1)
      
      # draw the histogram with the specified number of bins
      hist(x, breaks = bins, col = 'darkgray', border = 'white')
   })
}

# Run the application 
shinyApp(ui = ui, server = server)


```

### PR DESCRIPTION
**Pull request by wch**
This fixes #2245.

Test application:

```R
library(shiny)

ui <- fluidPage(
  actionButton('update','Update selectize inputs'),
  selectInput("select", "Original label (selectInput)", choices=c("A", "B", "C"), selectize = FALSE),
  selectizeInput("selectize", "Original label (selectizetInput)", choices=c("A", "B", "C"))
)

server <- function(input, output,session) {
  observeEvent(input$update,{
    updateSelectInput(session, "select", label="New label (selectInput)", choices = c("X", "Y", "Z"))
    updateSelectizeInput(session, "selectize", label="New label (selectizeInput)", choices = c("X", "Y", "Z"))
  })
}

shinyApp(ui = ui, server = server)
```

### CODE REVIEW

File: srcjs/input_binding_select.js [hunk 1]
```diff
@@ -22,10 +22,14 @@ $.extend(selectInputBinding, {
     return $(el).val();
   },
   setValue: function(el, value) {
-    var selectize = this._selectize(el);
-    if (typeof(selectize) !== 'undefined') {
-      selectize.setValue(value);
-    } else $(el).val(value);
+    if (this._is_selectize(el)) {
```
Comment (jcheng5) [inline srcjs/input_binding_select.js]: This conditional is backward

File: srcjs/input_binding_select.js [hunk 2]
```diff
@@ -119,8 +123,15 @@ $.extend(selectInputBinding, {
       this.setValue(el, data.value);
     }
 
-    if (data.hasOwnProperty('label'))
-      $(el).parent().parent().find('label[for="' + $escape(el.id) + '"]').text(data.label);
+    if (data.hasOwnProperty('label')) {
+      let label_name = $escape(el.id);
```
Comment (jcheng5) [inline srcjs/input_binding_select.js]: Better variable for `label_name` would be `escaped_id`

File: srcjs/input_binding_select.js [hunk 3]
```diff
@@ -141,6 +152,15 @@ $.extend(selectInputBinding, {
   initialize: function(el) {
     this._selectize(el);
   },
+  // Return true if it's a selectize input, false if it's a regular select input.
+  _is_selectize: function(el) {
+    var config = $(el).parent().find('script[data-for="' + $escape(el.id) + '"]');
+    if (config.length === 0) {
```
Comment (jcheng5) [inline srcjs/input_binding_select.js]: This could be simplified to `return config.length > 0;`

File: srcjs/input_binding_select.js (hunk 1 above)
Comment (wch) [reply srcjs/input_binding_select.js]: 😬

**Pull request merged**